### ISSUE TITLE
🐛 Bug Report: a11y - `EntityContextMenu` with no `extraContextMenuItems` has poor keyboard navigation

### ISSUE THREAD

**Issue by logonoff** [backstage/backstage#30399]
### 📜 Issue Labels

- [x] Please familiarize yourself with the issue labels used in this project: [LABELS.md](https://github.com/backstage/backstage/blob/master/LABELS.md)

### 🔎 Search Terms

```plain
EntityContextMenu, EntityHeader, EntityLayout
```

### 🗃️ Project Area

Catalog

### 🔗 External Integration

N/A

### 📝 Description & Context

When the `UNSTABLE_extraContextMenuItems` prop is not provided, the first menu item does not receive keyboard focus when the menu is open. When `UNSTABLE_extraContextMenuItems` is an empty array, a divider shows up when there are no items. 

### 👍 Expected Behavior

The divider is hidden when `UNSTABLE_extraContextMenuItems` is an empty array, and the "unregister entity" menu item is focused when the menu is open when no `extraContextMenuItems` are passed

### Have you read the Code of Conduct?

- [x] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes, and I have enough information to get started

**Comment by MihaiTabara** [comment]
Found in today's triaging meeting.
@logonoff thanks for filing this issue! 

Are you open to contributions to the project and if so, how might we help you bootstrapping in it?  I see you marked the last time as having enough information. Want me to assign this to you? Thank you!

**Comment by MihaiTabara** [comment]
Ongoing PR reviews, no stale.
